server/s3_part_metadata.cc

```diff
@@ -196,14 +197,32 @@ void S3PartMetadata::load_successful() {
   if (this->from_json(motr_kv_reader->get_value()) != 0) {
     s3_log(S3_LOG_ERROR, request_id,
            "Json Parsing failed. Index oid = "
-           "%" SCNx64 " : %" SCNx64 ", Key = %s, Value = %s\n",
+           "%" SCNx64 ":%" SCNx64 ", Key = %s, Value = %s\n",
            part_index_name_oid.u_hi, part_index_name_oid.u_lo,
            str_part_num.c_str(), motr_kv_reader->get_value().c_str());
     s3_iem(LOG_ERR, S3_IEM_METADATA_CORRUPTED, S3_IEM_METADATA_CORRUPTED_STR,
            S3_IEM_METADATA_CORRUPTED_JSON);
 
     json_parsing_error = true;
     load_failed();
+  } else if (!request->validate_attrs(bucket_name, object_name)) {
```

Comment:
anyways I propose to return back to the first version of the patch having in mind the following change is s3_iem:

diff --git a/server/s3_iem.h b/server/s3_iem.h
index 0850f958..809c3512 100644
--- a/server/s3_iem.h
+++ b/server/s3_iem.h
@@ -34,22 +34,30 @@ extern S3Option* g_option_instance;
 // 2. For each IEM: timestamp, nodename, pid, filename & line is included in
 //    the JSON data format by default. If IEM event has any additional data to
 //    send, then pass it as `json_fmt` param.
-#define s3_iem(loglevel, event_code, event_description, json_fmt, ...)      \
-  do {                                                                      \
-    std::string timestamp = s3_get_timestamp();                             \
-    s3_log(S3_LOG_INFO, "",                                                 \
-           "IEC: " event_code ": " event_description                        \
-           ": { \"time\": \"%s\", \"node\": \"%s\", \"pid\": "              \
-           "%d, \"file\": \"%s\", \"line\": %d" json_fmt " }",              \
-           timestamp.c_str(), g_option_instance->get_s3_nodename().c_str(), \
-           getpid(), __FILE__, __LINE__, ##__VA_ARGS__);                    \
-    if (loglevel == LOG_ALERT) {                                            \
-      s3_syslog(loglevel, "IEC:AS" event_code ":" event_description);       \
-    } else {                                                                \
-      s3_syslog(LOG_ERR, "IEC:ES" event_code ":" event_description);        \
-    }                                                                       \
+#define s3_iem_(loglevel, s3_loglevel, event_code, event_desc, json_fmt, ...) \
+  do {                                                                        \
+    if (loglevel == LOG_ALERT) {                                              \
+      s3_syslog(loglevel, "IEC:AS" event_code ":" event_desc);                \
+    } else {                                                                  \
+      s3_syslog(LOG_ERR, "IEC:ES" event_code ":" event_desc);                 \
+    }                                                                         \
+    std::string timestamp = s3_get_timestamp();                               \
+    s3_log(S3_ ## s3_loglevel, "",                                            \
+           "IEC: " event_code ": " event_desc                                 \
+           ": { \"time\": \"%s\", \"node\": \"%s\", \"pid\": "                \
+           "%d, \"file\": \"%s\", \"line\": %d" json_fmt " }",                \
+           timestamp.c_str(), g_option_instance->get_s3_nodename().c_str(),   \
+           getpid(), __FILE__, __LINE__, ##__VA_ARGS__);                      \
   } while (0)
+#define s3_iem(loglevel, event_code, event_desc, json_fmt, ...) \
+  s3_iem_(loglevel, LOG_INFO, event_code, event_desc, json_fmt, ##__VA_ARGS__)
+
+#define s3_iem_fatal(loglevel, event_code, event_desc, json_fmt, ...) \
+  s3_iem_(loglevel, LOG_ERROR, event_code, event_desc, json_fmt, ##__VA_ARGS__)

Reply:
moving s3_logging after iem allert logging will make s3_iem() semantics same as s3_log().

Reply:
Discussed Ivan/Anatoli -- decision: do not modify IEM implementation, do not remove logging that Dmitry added.  All is ok as it is in the patch now.  Resolving.